src/query/storage/m3/dynamic_cluster.go

```diff
@@ -60,16 +61,14 @@ type dynamicCluster struct {
 
 // NewDynamicClusters creates an implementation of the Clusters interface
 // supports dynamic updating of cluster namespaces.
-func NewDynamicClusters(_ DynamicClusterOptions) (Clusters, error) {
-	return nil, errors.New("dynamic cluster configuration not yet supported")
+func NewDynamicClusters(opts DynamicClusterOptions) (Clusters, error) {
+	return newDynamicClusters(opts, nil)
 }
 
-// TODO(nate): Replace constructor above with this method
-// once namespace staging is complete.
-//
-// newDynamicClusters creates an implementation of the Clusters interface
-// supports dynamic updating of cluster namespaces.
-func newDynamicClusters(opts DynamicClusterOptions) (Clusters, error) {
+// newDynamicClusters is a private constructor with an updater channel that can be
+// used to receive update notifications. This is should only be invoked directly by
+// tests.
```

Comment:
Hm, not a big fan of making methods more complex for the sake of testability; will add a few comments to the test for alternate approaches here